x/lockup/types/msgs.go

```diff
@@ -38,6 +38,10 @@ func (m MsgLockTokens) ValidateBasic() error {
 		return fmt.Errorf("lockups can only have one denom per lock ID, got %v", m.Coins)
 	}
 
+	if !m.Coins.IsAllPositive() {
```

Comment:
A test already exists for locking with no coins here https://github.com/osmosis-labs/osmosis/blob/344de9baa4db1a83b8eb3804ae4bdc1d28fe2dfe/x/lockup/keeper/lock_test.go#L282-L289

Should we have tests for ValidateBasic?

Reply:
Yes, I meant directly for `ValidateBasic`. 

Ideally, each method should have a corresponding unit test, including basic getters and setters.

For example, say we have an imaginary function:
```
func getPoolName() {
    return "osmo-pool"
}
```

Then, someone accidentally changes the return to "osmo-pul" while refactoring. Without a direct test, we might not catch it immediately and only see it in prod. On the other hand, having a direct test makes it helpful to catch such typos right away saving lots of time.

I don't think it's reasonable for us to come back and create such tests for everything. However, I think it would be great to keep adding tests for all newly added logic.

Let me know what you think

Reply:
Agreed, validate basic for messags should have table driven unit tests

Reply:
We actually should have done this as part of our audit, oof that we missed it

Reply:
I just wrote one for MsgLockTokens, I will make an issue to finish the rest!